src/main/java/com/schedule/workout/workoutScheduler/controller/user/UsersController.java

```diff
@@ -0,0 +1,96 @@
+package com.schedule.workout.workoutScheduler.controller.user;
+
+
+import com.schedule.workout.workoutScheduler.database.model.UserDB;
+import com.schedule.workout.workoutScheduler.exceptions.InvalidUserBodyException;
+import com.schedule.workout.workoutScheduler.exceptions.UserNotFoundException;
+import com.schedule.workout.workoutScheduler.services.UsersService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RestController
+public class UsersController {
+
+    @Autowired
+    private UsersService usersService;
+
+
+    @RequestMapping(method = RequestMethod.POST, value = "/users")
+    public ResponseEntity<UserDB> createUser(@Valid @RequestBody CreateUserModel createUserModel) {
+            try {
+                return ResponseEntity
+                        .status(HttpStatus.CREATED)
+                        .body(usersService.createUser(createUserModel));
+            }catch (InvalidUserBodyException e){
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                .build();
+        }
+    }
+
+    @RequestMapping(method = RequestMethod.PUT, value = "/users/{id}")
+    public ResponseEntity<UserDB> updateUser(@PathVariable String id,@Valid @RequestBody UpdateUserModel updateUserModel) {
+        try {
+            return ResponseEntity
+                    .status(HttpStatus.NO_CONTENT)
+                    .body(usersService.updateUser(id, updateUserModel));
+
+        } catch (UserNotFoundException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                    .build();
+        }catch (InvalidUserBodyException e){
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                .build();
+        }
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = "/users/{userModelId}")
+    public ResponseEntity<UserDB> getUserById(@PathVariable String userModelId) {
+        try {
+            return ResponseEntity
+                    .status(HttpStatus.OK)
+                    .body(usersService.getUserById(userModelId));
+        } catch (UserNotFoundException e) {
+            return ResponseEntity.status(HttpStatus.NOT_FOUND)
+                    .build();
+        }
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = "/users")
+    public ResponseEntity<List<UserDB>> getAllUsers(@RequestParam(name = "firstName", required = false) String firstName,
+                                                    @RequestParam(name = "lastName", required = false) String lastName) {
+        if (lastName != null){
```

Comment:
All of this if/else logic should be in service. Controller is only for calling service methods.

Also, add ability to filter users for both firstName and lastName.